language-antlr-utils/src/main/java/de/jplag/antlr/SemanticsDataHolder.java

```diff
@@ -0,0 +1,10 @@
+package de.jplag.antlr;
+
+import de.jplag.semantics.VariableRegistry;
+
+/**
+ * Holds all required context objects for the generation of code semantics
+ * @param registry The variable registry
+ */
+public record SemanticsDataHolder(VariableRegistry registry) {
```

Comment:
If this is a record encapsulating a single object, why do we need the record at all?

Reply:
Makes sense, but could be considered premature optimization of sorts. I am unsure if we should keep it in or not. @mbrdl do you have a preference?

Reply:
The variable registry even started off as a semantics data holder, the name only came later.

Reply:
@TwoOfTwelve then let us remove it for now! 👍